### PR TITLE
Aev/mfa update

### PR DESCRIPTION
**Pull request by avapirev**
changed pages:

mfa login
mfa quickstart
mfa guide

### CODE REVIEW

File: source/access/mfa_guide.rst [hunk 1]
```diff
@@ -3,6 +3,27 @@
...
+-------------------
+
+Mutli-Factor Authentication (MFA) is mandatory when connecting to the VSC network.
+In the context of SSH connections it sets an increased level of security by requiring
+an additional verification step. MFA ensures that even if an attacker obtains your SSH login
+credentials, such as username and password, they won’t be able to access the VSC netowrk
+without that additional authentication factor.
+
+Currently, there are two ways to connect to VSC: via the `VSC firewall page`_ and via
+SSH certificates agent. The firewall page method can be used without the need to set SSH
```
Comment (moravveji) [inline source/access/mfa_guide.rst:16]: and via SSH certificates.

File: source/access/mfa_guide.rst [hunk 2]
```diff
@@ -11,84 +32,45 @@ and you want to log in to the cluster with an ssh client like PuTTY or
...
-   keytype for connections to the HPC clusters, this should not be an issue for
-   most users.
+Users can also, upon establishing an SSH connection, store the SSH certificate in an SSH agent.
+The VSC firewall and insiitutional login steps described above are also valid here.
+The difference is that the user will be required to perform them only once in order to 
+set an active SSH certificate which will be then stored in the running SSH agent.
+Here are the necessary steps:
 
-.. _mfa_agent:
+#. Make sure the agent is running.
```
Comment (moravveji) [inline source/access/mfa_guide.rst:61]: Pleas add a hyperlink to the internal page about ssh agents.

File: source/access/mfa_guide.rst [hunk 3]
```diff
@@ -208,27 +195,61 @@ on your local machine first.
...
 
+GUI applications with SSH connection in the background
+------------------------------------------------------
+
+Some applications such as MobaXTerm and FileZilla provide a Graphical User Interface
+(GUI) which makes them very useful when connecting to remote sites. However,
+such an application may not always prompt you to copy/paste the VSC firewall link to
+set up the necessary SSH certificate.
+
+Therefore, one way to connect to VSC is to first connect with an ssh-client
```
Comment (moravveji) [inline source/access/mfa_guide.rst:206]: This is a bit ambiguous:
Where shall they first connect to (before connecting to VSC)? Please give a clear host address or extra explanation.

File: source/access/mfa_guide.rst [hunk 4]
```diff
@@ -208,27 +195,61 @@ on your local machine first.
...
+on your machine as explained in the previous sections.
+In the case of not having an SSH certificate agent running then as long as you
+keep that connection open you can connect with the other apps as well.
+In case you have already stored the SSH certificate in a running agent
+you can then proceed with connecting to VSC with the application.
+
+.. note::
+
+   Some GUI applications may not always work when connecting to VSC
+   without an agent. For example, NX is one of them.
```
Comment (moravveji) [inline source/access/mfa_guide.rst:216]: NX -> NoMachine
And please add a link to the internal page on NoMachine client setup

File: source/access/mfa_guide.rst [hunk 5]
```diff
@@ -98,7 +80,12 @@ together with PuTTY. Make sure you install version 0.78 or later.
 
 Pageant stores the certificates together with your private SSH keys, which allows you
 to use VSC facilities whenever you are prompted for your identity.
-To setup your Pageant, please refer to :ref:`Using Pageant <using Pageant>`.
+To setup your Pageant, please refer to :ref:`Using Pageant <using Pageant>` in
+the :ref:`mfa quick start`, or go to the respecitive section further below :ref:`mfa_client_config`.
+Reminder: in order to store the session SSH certificate make sure Pageant
+is already up and running in the background before attempting a conneciton with Putty.
+
+.. _mfa_agent_nix:
```
Comment (moravveji) [inline source/access/mfa_guide.rst:88]: In line 120, there is this sentence hanging in the air, and it has a very wrong formatting:

"(to kill the agent use ``eval "$(ssh-agent -k)"``)"

File: source/access/mfa_guide.rst [hunk 6]
```diff
@@ -208,27 +195,61 @@ on your local machine first.
...
 able to use an agent, you might have to make some changes in your connection
-profiles. Different apps will need different changes, but here we shortly show
-what to do for MobaXTerm, PuTTY and NX:
+profiles. Different apps will need different changes. For an explanation
+with screenshots for the various GUI apps please refer to :ref:`mfa quick start`.
+Here below we shortly show what to do for PuTTY, MobaXTerm, NX, and FileZilla:
+
+- PuTTY
+
+  - Under 'Session' in the tree menu enter the remote hostname. The port is usually 22.
```
Comment (moravveji) [inline source/access/mfa_guide.rst:231]: The default port is 22.

File: source/access/mfa_guide.rst [hunk 7]
```diff
@@ -238,10 +259,20 @@ what to do for MobaXTerm, PuTTY and NX:
   - Select 'Use key-based authentication with a SSH agent'
   - Click 'Modify' and verify that 'Forward authentication' is checked
 
+- FileZilla
```
Comment (moravveji) [inline source/access/mfa_guide.rst:262]: For FileZilla, we do have a dedicated page: https://docs.vscentrum.be/data/transfer/filezilla.html
I strongly recommend to remove the text here and only refer to that page. Else, we shall be maintaining two separate pages!

File: source/access/mfa_guide.rst [hunk 8]
```diff
@@ -208,27 +195,61 @@ on your local machine first.
...
+
+  - Under 'Session' in the tree menu enter the remote hostname. The port is usually 22.
+  - Under 'Connection/SSH' ensure the protocol is set to SSH.
+  - Under 'Connection/SSH/Auth' ensure that using Pageant and agent forwarding are active.
+  - Under 'Connection/SSH/Auth/Credentials' make sure that there are no private keys specified
+    in the 'Private key file for authentication' field.
+  - Go back to 'Session' and save your newly created session.
+  - Now you can load and open it to connect to VSC.
 
 - MobaXTerm
```
Comment (moravveji) [inline source/access/mfa_guide.rst:239]: There is a dedicated page for MobaXterm: https://docs.vscentrum.be/access/access_using_mobaxterm.html
I have not gone through that page (specially at the bottom where it talks about Moba's internal SSH agent - being Pageant). but, maybe what you have here is already there. Best is to refer to that page, if it covers our MFA case.

File: source/access/mfa_guide.rst [hunk 9]
```diff
@@ -208,27 +195,61 @@ on your local machine first.
...
 
 If you have not yet set up your ssh-client or other apps that use ssh to be
 able to use an agent, you might have to make some changes in your connection
-profiles. Different apps will need different changes, but here we shortly show
-what to do for MobaXTerm, PuTTY and NX:
+profiles. Different apps will need different changes. For an explanation
+with screenshots for the various GUI apps please refer to :ref:`mfa quick start`.
+Here below we shortly show what to do for PuTTY, MobaXTerm, NX, and FileZilla:
+
+- PuTTY
```
Comment (moravveji) [inline source/access/mfa_guide.rst:229]: The PuTTY page (https://docs.vscentrum.be/access/text_mode_access_using_putty.html) unfortunately does not talk about Certificates, though it covers agents. 
I guess, at some point in near future, we have to take an initiative to update those pages, and only provide a link to those pages in this MFA document. 
But, this part is OK for now.

**Review by moravveji — CHANGES_REQUESTED** [review]
Thanks Alex for taking up reviewing this page. It really deserved a good polish.

I have partially read through the first file (mfa_guide.rst), and have put my comments.
They are easy to fix. I will continue later with the rest.

File: source/access/mfa_guide.rst [hunk 10]
```diff
@@ -3,6 +3,27 @@
 MFA Guide
 =========
 
+General information
+-------------------
+
+Mutli-Factor Authentication (MFA) is mandatory when connecting to the VSC network.
```
Comment (stevenvdb) [inline source/access/mfa_guide.rst:9]: Mutli -> Multi

Also note that this is not enforced everywhere, so perhaps better to use the "|KUL|" tag for now?

File: source/access/mfa_guide.rst (hunk 7 above)
Comment (avapirev) [reply source/access/mfa_guide.rst:262]: This one you take with Maxime. I follow what he and Wouter have discussed. For now I will leave it.

File: source/access/mfa_guide.rst (hunk 8 above)
Comment (avapirev) [reply source/access/mfa_guide.rst:237]: Yes, same situation here as with filezilla. In general, a lot of connection/mfa pages need to be revisited. We only need to make sure we do not have wrong info for Tuesday.

File: source/access/mfa_guide.rst (hunk 9 above)
Comment (avapirev) [reply source/access/mfa_guide.rst:227]: Oh, I suppose I could also refer to that page. Later we review.

File: source/access/mfa_guide.rst (hunk 5 above)
Comment (avapirev) [reply source/access/mfa_guide.rst:88]: indeed, removed.

File: source/access/mfa_guide.rst (hunk 8 above)
Comment (avapirev) [reply source/access/mfa_guide.rst:237]: The issue with that page is that it still talks about SSH keys. Same with the Putty pages. I will only put tag KUL for now.

File: source/access/mfa_guide.rst (hunk 9 above)
Comment (avapirev) [reply source/access/mfa_guide.rst:227]: I prefer that we do not point to that page for now. It shows SSH keys and people will get confused.

File: source/access/mfa_guide.rst (hunk 3 above)
Comment (avapirev) [reply source/access/mfa_guide.rst:206]: I think now it's better

**Pull request closed without merging**